source/elements/oneDAL/include/oneapi/dal/array.hpp

```diff
@@ -0,0 +1,290 @@
+namespace oneapi::dal {
+
+/// @tparam Data The type of the memory block elements within the array.
+///
+/// @pre $Data$ cannot be const-qualified.
+template <typename Data>
+class array {
+public:
+    using data_t = Data;
+
+public:
+    /// Allocates a new memory block for mutable data, does not initialize it,
+    /// creates a new array instance by passing a pointer to memory block.
```

Comment:
if that's the same memory block mentioned in the beginning 
```suggestion
    /// creates a new array instance by passing a pointer to the memory block.
```

applies in other places as well